### ISSUE TITLE
Add capability of specifying PrefferedAudioGroups for Stop/Rank/Pipe in ODF

### ISSUE THREAD

**Issue by oleg68** [GrandOrgue/grandorgue#732]
Most of sample sets are multichannel now. Usually they have front and rear samples. Sometimes there are other channels.

When the GrandOrgue setup has several audio speakers and audio groups, it requires a large amount of work to assign ranks or even pipes to audio groups.

It would be better to have preffered audio group names specified in the ODF for ranks, stops and for pipes. If the audio group exists with the name in the list, it will be assigned automatically

**Comment by larspalo** [comment]
I'm not at all sure this is a good idea. The end user should always have the final word to adapt any sampleset to be used on his/her particular setup. Audio groups are created by the user depending on the setup in question. The sampleset producer should not have any possibility to enforce how the audio groups should be routed or called.

Thus, while it might seem that this is making things easier for the user, it's actually likely to create more problems than it will ever solve.

Also remember that the audio groups can be used for completely different reasons than multichannel samples. Flexibility, which is one of the real strengths of GO, will always come at the price of having the responsibility to learn how to use it.

**Comment by oleg68** [comment]
@larspalo I agree that specifying audio groups is not a very good idea of implementation. But the problem still exists:

1. Almost all modern sample sets are multi-channel.
2. The right distributions of audio channels requires a large amount of work.

I renamed the title. May be to think on a better idea how to make this work easier.

One anoter idea might be to introduce the AudioChannel object in ODF. Sample set creators whould be able to map ranks to AudioChannels, and users could map between AudioChannels in ODF and AudioGroups in GO on the whole organ level, on the windchest level, or on the rank level with capability of direct assigning audio groups to individual pipes.

**Comment by larspalo** [comment]
Ok, this already exist. In the ODF it's called WindchestGroup999. It just requires a little planning and knowledge on part of both sampleset creator and user.

If the sampleset creator (or ODF maker) is kind he will put the different ranks (front, rear, side, whatever) on separate windchests that's clearly named such that the user can identify what their purpose is. Thus you can have, in a simple 4 ch recording, a "Front Great" windchest and a "Rear Great" windchest. Then it's very easy for the user to, after having created the relevant audio groups of course for his/her device, just distributing the corresponding windchests to the correct groups.

This might seem like a deviation from the true organ model, but if you look at it from the different microphones placement point of view then it starts to make sense. In a front/rear setup the microphones will hear different views of the organ. You could say that it's a little like with mirrors. The front microphones will hear one side and the rear will hear another.

Then again, we should always remember that the audio groups can be used in other creative ways too that's not limited to only multichannel recordings/setups.

**Comment by oleg68** [comment]
> If the sampleset creator (or ODF maker) is kind he will put the different ranks (front, rear, side, whatever) on separate windchests that's clearly named such that the user can identify what their purpose is. Thus you can have, in a simple 4 ch recording, a "Front Great" windchest and a "Rear Great" windchest. Then it's very easy for the user to, after having created the relevant audio groups of course for his/her device, just distributing the corresponding windchests to the correct groups.

Iy is a good idea. I'll test it.

> This might seem like a deviation from the true organ model

Having several ranks (front, rear, front+tremulant, rear+tremulant) for the same physical pipe group is also a deviation.

Maybe to think on introducing two level ranks and windchest - physical and virtual?

**Comment by larspalo** [comment]
> Having several ranks (front, rear, front+tremulant, rear+tremulant) for the same physical pipe group is also a deviation.

Yes, but that's because the sample is recorded with multiple microphone pairs in different places. As a matter of fact - this is not how the human ear will hear the organ. This is just a construct to make it easier to understand what microphones were used to record the sample, and in the end how it's supposed to be played back.
